Update observability doc structure and content (#35816)
ray-project/ray: doc/source/ray-observability/user-guides/debug-apps/optimize-performance.rst

Visualizing Tasks in the Ray Timeline
-------------------------------------

> 1. The :ref:`timeline API <ray-core-timeline>` is available from Ray Dashboard.

Reviewer: Add a link to the dashboard page timeline?

Author: Done

> * First, you can download the chrome tracing file by clicking the download button.

Reviewer 2: ```suggestion
* Download the Chrome tracing file by clicking the download button.
```

> * Second, you can use tools like ``chrome://tracing`` or the `Perfetto UI <https://ui.perfetto.dev/>`_ and drop the downloaded chrome tracing file. We will use the Perfetto as it is the recommendation way to visualize chrome tracing files.

Reviewer 2: ```suggestion
* Second, you can use tools like ``chrome://tracing`` or the `Perfetto UI <https://ui.perfetto.dev/>`_ and drop the downloaded chrome tracing file. Using the Perfetto UI is the recommended way to visualize chrome tracing files.
```

> * Second, you can use tools like ``chrome://tracing`` or the `Perfetto UI <https://ui.perfetto.dev/>`_ and drop the downloaded chrome tracing file. We will use the Perfetto as it is the recommendation way to visualize chrome tracing files.

Reviewer 2: ```suggestion
* Use tools like ``chrome://tracing`` or the `Perfetto UI <https://ui.perfetto.dev/>`_ and drop the downloaded Chrome tracing file. Use the Perfetto as it is the recommended way to visualize Chrome tracing files.
```

> * Now, you can see the timeline visualization of Ray tasks and actors. There are Node rows (hardware) and Worker rows (processes). Each worker rows display a list of events (e.g., task scheduled, task running, input/output deserialization, etc.) happening from that worker over time.

Reviewer 2: ```suggestion
* Now, you can see the timeline visualization of Ray tasks and actors. There are Node rows (hardware) and Worker rows (processes). Each worker row displays a list of events (e.g., Task scheduled, Task running, input/output deserialization, etc.) happening from that worker over time.
```

> * Now, you can see the timeline visualization of Ray tasks and actors. There are Node rows (hardware) and Worker rows (processes). Each worker rows display a list of events (e.g., task scheduled, task running, input/output deserialization, etc.) happening from that worker over time.

Reviewer 2: ```suggestion
* See the timeline visualization of Ray Tasks and Actors. There are Node rows (hardware) and Worker rows (processes). Each Worker row displays a list of events (e.g., Task scheduled, Task running, input/output deserialization, etc.) happening from that Worker over time.
```

> 2. You can also dump the tracing file as a JSON file by running ``ray timeline`` from the command line or ``ray.timeline`` from the Python API.

Reviewer 2: ```suggestion
2. You can also export the tracing file as a JSON file by running ``ray timeline`` from the command line or ``ray.timeline`` from the Python API.
```

.. code-block:: python
ray.timeline(filename="/tmp/timeline.json")

> To use the timeline, enable Ray profiling by setting the ``RAY_PROFILING=1`` environment variable prior to starting Ray on every machine, and setting the ``RAY_task_events_report_interval_ms`` to larger than 0 (default 1000).

Reviewer: Both are on by default, so we can remove it I think cc @ricky to confirm.

Author: Removed this for now.

> To use the timeline, Ray profiling must be enabled by setting the ``RAY_PROFILING=1`` environment variable prior to starting Ray on every machine, and ``RAY_task_events_report_interval_ms`` must be larger than 0 (default 1000).

Reviewer 2: ```suggestion
To use the timeline, enable Ray profiling by setting the ``RAY_PROFILING=1`` environment variable prior to starting Ray on every machine, and setting the ``RAY_task_events_report_interval_ms`` to larger than 0 (default 1000).
```

  20    0.001    0.000    0.001    0.000 worker.py:514(submit_task)
  ...

GPU Profiling
------------------------

> Ray currently doesn't provide native integration with GPU profiling tools. Try running GPU profilers like PyTorch Profiler without Ray to identify the issues.

Reviewer 2: ```suggestion
Ray doesn't provide native integration with GPU profiling tools. Try running GPU profilers like PyTorch Profiler without Ray to identify the issues.
```

> If you have related feature requests, please `let us know`_.

Reviewer 2: ```suggestion
If you have related feature requests, `let us know`_.
```
